test/unit/data-type.js

```diff
@@ -75,9 +75,866 @@ describe('Data Types', function() {
+
+    const type = TYPES.typeByName.Binary;
+
+    const buffer = new WritableTrackingBuffer(0);
+    const parameterValue = { value, length: 4 };
+
+    type.writeParameterData(buffer, parameterValue, { useUTC: false }, () => {
+      assert.deepEqual(buffer.data, expected);
+
+      done();
+    });
+  });
+
+  it('should writeParameterData Bit (Buffer)', function(done) {
+    const value = 1;
+    const expected = Buffer.from('0101', 'hex');
+
+    const type = TYPES.typeByName.Bit;
+
+    const buffer = new WritableTrackingBuffer(0);
+    const parameterValue = { value };
+
+    type.writeParameterData(buffer, parameterValue, { useUTC: false }, () => {
+      assert.deepEqual(buffer.data, expected);
+      done();
+    });
+  });
+
+  it('should writeParameterData Bit (Null)', function(done) {
+    const value = null;
+    const expected = Buffer.from('00', 'hex');
+
+    const type = TYPES.typeByName.Bit;
+
+    const buffer = new WritableTrackingBuffer(0);
+    const parameterValue = { value };
+
+    type.writeParameterData(buffer, parameterValue, { useUTC: false }, () => {
+      assert.deepEqual(buffer.data, expected);
+      done();
+    });
+  });
+
+  it('should writeParameterData Bit (Undefined)', function(done) {
+    const value = undefined;
+    const expected = Buffer.from('00', 'hex');
+
+    const type = TYPES.typeByName.Bit;
+
+    const buffer = new WritableTrackingBuffer(0);
+    const parameterValue = { value };
+
+    type.writeParameterData(buffer, parameterValue, { useUTC: false }, () => {
+      assert.deepEqual(buffer.data, expected);
+      done();
+    });
+  });
+
+  it('should writeParameterData Char (Buffer)', function(done) {
+    const value = Buffer.from([0xff, 0xff, 0xff, 0xff]);
+    const expected =  Buffer.from('0400ffffffff', 'hex');
+
+    const type = TYPES.typeByName.Char;
+
+    const buffer = new WritableTrackingBuffer(0);
+    const parameterValue = { value };
+
+    type.writeParameterData(buffer, parameterValue, { useUTC: false }, () => {
+      assert.deepEqual(buffer.data, expected);
+      done();
+    });
+  });
+
+  it('should writeParameterData Char (Null)', function(done) {
+    const value = null;
+    const expected = Buffer.from('ffff', 'hex');
+
+    const type = TYPES.typeByName.Char;
+
+    const buffer = new WritableTrackingBuffer(0);
+    const parameterValue = { value };
+
+    type.writeParameterData(buffer, parameterValue, { useUTC: false }, () => {
+      assert.deepEqual(buffer.data, expected);
+      done();
+    });
+  });
+  
+  it('should writeParameterData DateTimeOffSet (Buffer)', function(done) {
+    const value = new Date(Date.UTC(2014, 1, 14, 17, 59, 59, 999));
+    const expected = Buffer.from('0820fd002d380b20fe', 'hex');
+
+    const type = TYPES.typeByName.DateTimeOffset;
+
+    const buffer = new WritableTrackingBuffer(0);
+    const parameterValue = { value, scale: 0 };
+
+    type.writeParameterData(buffer, parameterValue, { useUTC: false }, () => {
```

Comment:
I think this test is failing because `useUTC: false` instead of `useUTC: true` here.

The `expected` value is based on what the value would be in your timezone, while the system running the tests might be in a different timezone and thus sending a different value and offset.

Try changing `useUTC` to `true` and update the `expected` value to be UTC based (offset = 0), and then the tests should also pass on CI and other systems.

Reply:
Slicing the last two buffers off, which holds different values based on local machine timezone